core/state/access_witness.go

```diff
@@ -284,7 +284,7 @@ func (aw *AccessWitness) TouchBasicData(addr []byte, isWrite bool, useGasFn UseG
 	return ok && (!warmCostCharging || chargedGas > 0 || useGasFn(params.WarmStorageReadCostEIP2929))
 }
 
-func (aw *AccessWitness) TouchCodeHash(addr []byte, isWrite bool, useGasFn UseGasFn) bool {
+func (aw *AccessWitness) TouchCodeHash(addr []byte, isWrite bool, useGasFn UseGasFn, warmCostCharging bool) bool {
```

Comment:
The previous version always charged WARM cost. For the case we were fixing yesterday, where we include CODEHASH as intrinsic cost, this meant we were still charging 100 of gas (WARM).

With this change we allow this function to be signaled if we want (or not) to charge warm costs. There're two usages of this method: the one we were discussing yesterday, and EXTCODEHASH. For the former, warm cost is disabled. For the latter, it's enabled.